### PR TITLE
Support variable number of elements in common hieradata

### PR DESCRIPTION
**Pull request by AlbertoPeon**
This PR solves #3

### CODE REVIEW

File: src/jens/test/update.py [hunk 1]
```diff
@@ -73,6 +73,18 @@ def test_base(self):
         self.assertEnvironmentLinks("qa")
         self.assertEnvironmentLinks("production")
 
+    def test_base_empty_common_hieradata(self):
```
Comment (nbarrientos) [inline src/jens/test/update.py]: I'd change the test name to "test_empty_common_hieradata" and...

File: src/jens/test/update.py [hunk 2]
```diff
@@ -73,6 +73,18 @@ def test_base(self):
...
+        self.settings.HIERADATA_KEYS = []
+        self._jens_update()
+
+        self.assertBare('common/site')
```
Comment (nbarrientos) [inline src/jens/test/update.py]: ...remove statements from line 80 to 84 as that's covered by test_base.

File: src/jens/test/update.py [hunk 3]
```diff
@@ -73,6 +73,18 @@ def test_base(self):
...
+        self.assertBare('common/hieradata')
+        for branch in MANDATORY_BRANCHES:
+            self.assertClone('common/site/%s' % branch)
+            self.assertClone('common/hieradata/%s' % branch)
+        self.assertEnvironmentLinks("qa")
+        self.assertEnvironmentLinks("production")
+
```
Comment (nbarrientos) [inline src/jens/test/update.py]: What about adding the following test?

```
+    def test_missing_common_hieradata_item(self):
+        self.settings.HIERADATA_KEYS = ["missing"]
+        self._jens_update()
+
+        self.assertEnvironmentBrokenLinks("qa")
+        self.assertEnvironmentBrokenLinks("production")
+
```

File: src/jens/configfile.py [hunk 4]
```diff
@@ -19,6 +19,7 @@
 cachedir = string(default='/var/lib/jens/cache')
 hashprefix = string(default='commit/')
 directory_environments = boolean(default=False)
+hieradata_keys = list(default=list())
```
Comment (nbarrientos) [inline src/jens/configfile.py]: common_hieradata_items?

**Comment by AlbertoPeon** [comment]
Done, could you check again?

File: README.md [hunk 5]
```diff
@@ -119,8 +119,9 @@ it-puppet-site/
 
 OTOH, the common hieradata is a special repository that hosts different types
 of Hiera data to fill the gaps that can't be defined at hostgroup or module
-level (operating system, hardware vendor and environment dependent keys). The
-following is an example of how the hierarchy in there should look like.
+level (operating system, hardware vendor, datacentre location and environment
+dependent keys). The following is an example of how the hierarchy in there
+should look like.
```
Comment (nbarrientos) [inline README.md]: Probably useful to mention the new configuration key here somewhere.

File: src/jens/environments.py [hunk 6]
```diff
@@ -444,14 +444,13 @@ def _link_site(settings, environment, definition):
 def _link_common_hieradata(settings, environment, definition):
     # Global scoped (aka, 'common') Hiera data
     # LINK_NAME: $environment/hieradata/
-    # {environments, hardware, operatingsystems, common.yaml}
-    # TARGET: $clonedir/common/hieradata/$branch/code/{ditto}
+    # {environments, hardware, operatingsystems, datacentres, common.yaml}
```
Comment (nbarrientos) [inline src/jens/environments.py]: What about settings.COMMON_HIERADATA_ITEMS instead?

File: examples/hiera.yaml [hunk 7]
```diff
@@ -14,6 +14,7 @@
          - environments/%{::foreman_env}/hieradata/hostgroups/%{::encgroup_0}/%{::encgroup_0}/%{::encgroup_1}
          - environments/%{::foreman_env}/hieradata/hostgroups/%{::encgroup_0}/%{::encgroup_0}/operatingsystems/%{::osfamily}/%{::operatingsystemmajorrelease}
          - environments/%{::foreman_env}/hieradata/hostgroups/%{::encgroup_0}/%{::encgroup_0}
+         - environments/%{::foreman_env}/hieradata/datacentres/%{::datacentres}
```
Comment (nbarrientos) [inline examples/hiera.yaml]: s/%{::datacentres}/%{::datacentre}, as our fact is called 'datacentre'.

**Pull request merged**